provider/core/src/main/java/org/arquillian/pact/provider/core/InteractionRunner.java

```diff
@@ -125,6 +128,21 @@ private void setField(Object testInstance, Field fieldTarget, Object pact) {
         }
     }
 
+    protected void validateState(final TestClass testClass, final List<Throwable> errors) {
+        Arrays.stream(testClass.getMethods(State.class)).forEach(method -> {
+            validatePublicVoidMethods(method, errors);
+            if (method.getParameterCount() == 1 && !Map.class.isAssignableFrom(method.getParameterTypes()[0])) {
```

Comment:
If we open a PR I thought it will be reviewed by someone :)

How about extracting this condition to a meaningful method to describe what it's really checking?